### PR TITLE
feature/implementation pokemon details

### PR DESCRIPTION
**Pull request by verywellced**
**feat:  implementation of pokemon details**

- Add model for pokemon details
- Add model for pokemon moves
- Add model for pokemon abilities
- Add model for pokemon stats
- Add model for pokemon stat
- Add testing code on the main.dart

- Merged from updated review branch 

(PR Feedback)
-  Removed Prefixed on pokemon details sub models
-  Maximized Pokemon Details on pokemon api
-  Changed parameters on getting details from id to name 
-  Rerun flutter freezed on model

(PR Feedback)
-  Changed the line limit to 120
- Change some function to one liner'

Height, Base Experience and weight

Ability and Move

Statlist

Working app

### CODE REVIEW

File: lib/api/handler/pokemon_api.dart [hunk 1]
```diff
@@ -30,4 +34,46 @@ class PokemonApi {
...
+              (dynamic moves) => PokemonMove.fromJson(moves['move'] as Json))
+          .toList();
+
+      // Get the stats of pokemon
+      List<PokemonStats> pokemonStatList = jsonStats
+          .map<PokemonStats>(
+              (dynamic stats) => PokemonStats.fromJson(stats as Json))
+          .toList();
+
+      return PokemonDetails(
```
Comment (shawn-ffufm) [inline lib/api/handler/pokemon_api.dart]: why not maximize the PokemonDetails.fromJson instead to parse everything like the way we did for pokemon list? 🙂

for reference take a look at henry's pr if ever 🙂

File: lib/api/handler/pokemon_api.dart [hunk 2]
```diff
@@ -30,4 +34,46 @@ class PokemonApi {
           .toList();
     });
   }
+
+  Future<PokemonDetails> getPokemonDetails({
+    required String pokemonId,
```
Comment (shawn-ffufm) [inline lib/api/handler/pokemon_api.dart]: no need for extra trailing comma

Comment (shawn-ffufm) [inline lib/api/handler/pokemon_api.dart]: use pokemon name instead, it'll be much easier than using id

File: lib/api/model/pokemon_ability.dart [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'pokemon_ability.freezed.dart';
+part 'pokemon_ability.g.dart';
+
+@freezed
+class PokemonAbility with _$PokemonAbility {
+  factory PokemonAbility({
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_ability.dart]: kindly remove Pokemon prefix - Ability is already enough 🙂 

this also applies to some other models except for `pokemon_details.dart `, just leave it as is.

File: lib/api/model/pokemon_ability.dart [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    @JsonKey(name: 'name') required String name,
+    @JsonKey(name: 'url') required String url,
+  }) = _PokemonAbility;
+
+  factory PokemonAbility.fromJson(Map<String, dynamic> json) =>
+      _$PokemonAbilityFromJson(json);
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_ability.dart]: make this one liner instead 🙂

File: lib/api/model/pokemon_details.dart [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    @JsonKey(name: 'height') required int height,
+    @JsonKey(name: 'weight') required int weight,
+    @JsonKey(name: 'base_experience') required int baseExperience,
+    @JsonKey(name: 'abilities') required List<PokemonAbility> abilityList,
+    @JsonKey(name: 'moves') required List<PokemonMove> moveList,
+    @JsonKey(name: 'stats') required List<PokemonStats> statList,
+  }) = _PokemonDetails;
+
+  factory PokemonDetails.fromJson(Map<String, dynamic> json) =>
+      _$PokemonDetailsFromJson(json);
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_details.dart]: make this one liner as well

File: lib/api/model/pokemon_move.dart [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+@freezed
+class PokemonMove with _$PokemonMove {
+  factory PokemonMove({
+    @JsonKey(name: 'name') required String name,
+    @JsonKey(name: 'url') required String url,
+  }) = _PokemonMove;
+
+  factory PokemonMove.fromJson(Map<String, dynamic> json) =>
+      _$PokemonMoveFromJson(json);
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_move.dart]: make this one liner as well

**Review by shawn-ffufm — CHANGES_REQUESTED** [review]
see pr feedback 🙂

File: lib/api/model/pokemon_move.dart (hunk 6 above)
Comment (verywellced) [reply lib/api/model/pokemon_move.dart]: Hello Sir, it comes outside the vertical line limit. Should I remove the limit or adjust it?

File: lib/api/model/pokemon_details.dart (hunk 5 above)
Comment (verywellced) [reply lib/api/model/pokemon_details.dart]: Hello Sir, it comes outside the vertical line limit. Should I remove the limit or adjust it?

File: lib/api/model/pokemon_move.dart (hunk 6 above)
Comment (shawn-ffufm) [reply lib/api/model/pokemon_move.dart]: ayt, keep it as is then 👍

Comment (shawn-ffufm) [reply lib/api/model/pokemon_move.dart]: marking as resolved 🙂

**Review by shawn-ffufm — APPROVED** [review]
lgtm 💪

**Pull request merged**